### PR TITLE
PCSX2: Minor memory card fixes

### PR DESCRIPTION
**Pull request by FlatOutPS2**
Fixes a few minor issues in the memory card dialogs.

Notable changes:

Replaces the awkward checkbox for a PSX memory card in the bottom left
of the Create Memory Card dialog with a radio button like the other
memory card types.

Changes the file extension used by PSX memory cards to the common .mcr
instead of using the same (.ps2) extension used by PS2 memory cards.

### CODE REVIEW

File: pcsx2/gui/Panels/MemoryCardListPanel.cpp [hunk 1]
```diff
@@ -1016,29 +1014,27 @@ void Panels::MemoryCardListPanel_Simple::OnOpenItemContextMenu(wxListEvent& evt)
...
 		const McdSlotItem& card( GetCardForViewIndex(idx) );
 
-		if (card.IsPresent){
-			junk->Append( McdMenuId_AssignUnassign,	card.Slot>=0?_("Eject card"):_("Insert card ...") );
-			junk->Append( McdMenuId_Duplicate,	_("Duplicate card ...") );
-			junk->Append( McdMenuId_Rename,		_("Rename card ...") );
-			junk->Append( McdMenuId_Create,		_("Delete card") );
+		if (card.IsPresent) {
+			junk->Append( McdMenuId_AssignUnassign,	card.Slot>=0?_("Eject card"):_("&Insert card ...") );
+			junk->Append( McdMenuId_Duplicate,	_("D&uplicate card ...") );
```
Comment (ramapcsx2) [inline pcsx2/gui/Panels/MemoryCardListPanel.cpp]: Hey, what happened here?

Comment (FlatOutPS2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: Keyboard shortcuts.

Comment (ssakash) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: You can navigate via the options using the Alt key.

Comment (FlatOutPS2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: They're shortcuts for the right-click menu. ;)

Comment (ramapcsx2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: I don't get it. _("D&uplicate card ...") ) is a keyboard shortcut? Could the ampersand be any place more random? :p

Comment (FlatOutPS2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: There's another shortcut beginning with a d....

Comment (ramapcsx2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: I'll just file this under GUI coding adventures. Always something new and awesome to discover!

Comment (ramapcsx2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: The patch is fine by me, by the way. Good stuff :)

**Comment by turtleli** [comment]
IMO you should split the commit up into multiple smaller commits. There's no mention at all in the commit message about the menu keyboard shortcuts, and it's hard to see what's changed due to the formatting and other changes that are mixed into the commit.

File: pcsx2/gui/Panels/MemoryCardListPanel.cpp (hunk 1 above)
Comment (ssakash) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: Personally I suggest adding a new function for it to improve code readability. Something like ``MenuShortcut("Insert Card", "I")`` which returns the respective string with a shortcut letter of "I".

Any opinions? @FlatOutPS2 @turtleli

**Comment by ssakash** [comment]
> IMO you should split the commit up into multiple smaller commits. There's no mention at all in the commit message about the menu keyboard shortcuts, and it's hard to see what's changed due to the formatting and other changes that are mixed into the commit.

Is it really ideal to add new commits for trailing whitespaces removal, conditional check formatting changes, copyright date updates? (Those are the only non-relevant changes I can see and I personally don't find it hard to discern them from the actual changes)

**Comment by turtleli** [comment]
> Is it really ideal to add new commits for trailing whitespaces removal, conditional check formatting changes, copyright date updates? (Those are the only non-relevant changes I can see and I personally don't find it hard to discern them from the actual changes)

IMO it's better to do so if there's a lot of it (I don't mind copyright changes being thrown in wherever) and I think there's quite a bit in this case. It makes it a lot more obvious what the changes are, and when you're hunting down regressions you really do not want to end up staring at someone's "here's a little bit of everything" commit.

File: pcsx2/gui/Panels/MemoryCardListPanel.cpp (hunk 1 above)
Comment (turtleli) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: > Personally I suggest adding a new function for it to improve code readability. Something like MenuShortcut("Insert Card", "I") which returns the respective string with a shortcut letter of "I".

I don't think that'd work well with localisation.

Comment (ssakash) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: Yeah, it won't. Bad idea :P

**Comment by FlatOutPS2** [comment]
So how many commits do you want?
1. checkbox replace
2. mcr extension
3. keyboard shortcuts
4. code cleanup
?

Should I update the translatable strings as well?

**Comment by turtleli** [comment]
> So how many commits do you want?

Dunno.
2 - keyboard shortcuts
3+ - Up to you, I'd personally do at least 2 more (.mcr, GUI changes)
1 - Code formatting changes that don't belong in other commits.

> Should I update the translatable strings as well

If you mean the po/mo files - Probably no. I think there's a script that takes care of it, and the current .po files already have a few translatable strings missing.

File: pcsx2/gui/Dialogs/CreateMemoryCardDialog.cpp [hunk 2]
```diff
@@ -248,12 +255,14 @@ void Dialogs::CreateMemoryCardDialog::CreateControls()
...
 
-	m_radio_CardSize = new pxRadioPanel( this, tbl_CardSizes );
-	m_radio_CardSize->SetDefaultItem(0);
+		RadioPanelItem(_("128 KiB (PSX)"), _("This is the standard Sony-provisioned size PSX memory card, only compatible with PSX games."))
+		.	SetToolTip(_t("This memory card is required by PSX games. It is not compatible with PS2 games."))
+		.	SetInt(1)
+	};
 
-	m_check_psx = new pxCheckBox(this, "Make this a PSX card (128 KiB only, no folders!)");
+	m_radio_CardType = new pxRadioPanel( this, tbl_CardTypes);
```
Comment (turtleli) [inline pcsx2/gui/Dialogs/CreateMemoryCardDialog.cpp]: Spaces are inconsistent.

File: pcsx2/gui/Dialogs/CreateMemoryCardDialog.cpp [hunk 3]
```diff
@@ -164,7 +169,9 @@ void Dialogs::CreateMemoryCardDialog::OnOk_Click( wxCommandEvent& evt )
 	g_Conf->McdCompressNTFS = m_check_CompressNTFS->GetValue();
 #endif
 	result_createdMcdFilename=L"_INVALID_FILE_NAME_";
-	wxString composedName = m_text_filenameInput->GetValue().Trim() + L".ps2";
+
+	wxString composedName;
+	composedName = m_text_filenameInput->GetValue().Trim() + (m_radio_CardType->SelectedItem().SomeInt == 1 ? L".mcr" : L".ps2");
```
[the remark this answers is not in the document]
Comment (FlatOutPS2) [reply pcsx2/gui/Dialogs/CreateMemoryCardDialog.cpp]: Not sure if remnant of git messing with my PR or just something I missed. :p

File: pcsx2/gui/Panels/MemoryCardListPanel.cpp [hunk 4]
```diff
@@ -621,36 +616,38 @@ void Panels::MemoryCardListPanel_Simple::DoRefresh()
...
 	Apply();
 	RefreshSelections();
 }
 
-void Panels::MemoryCardListPanel_Simple::UiConvertCard( McdSlotItem& card ) {
-	if ( !card.IsPresent ) {
-		Console.WriteLn( "Error: Aborted: Convert mcd invoked but but a file is not associated." );
+void Panels::MemoryCardListPanel_Simple::UiConvertCard( McdSlotItem& card )
+{
+	if (!card.IsPresent ){
```
[the remark this answers is not in the document]
Comment (FlatOutPS2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: > IMO we really should just clang-format pcsx2 and gsdx ASAP

Go for it. :)

File: pcsx2/gui/Panels/MemoryCardListPanel.cpp [hunk 5]
```diff
@@ -1030,9 +1026,8 @@ void Panels::MemoryCardListPanel_Simple::OnOpenItemContextMenu(wxListEvent& evt)
 		} else {
 			junk->Append(McdMenuId_Create, _("&Create a new card ..."));
 		}
-		else
-			junk->Append( McdMenuId_Create, _("Create a new card ...") );
-
+	} else {
+		junk->Append(McdMenuId_Create, _("&Create a new card ..."));
```
Comment (turtleli) [inline pcsx2/gui/Panels/MemoryCardListPanel.cpp]: These changes belong to the menu shortcut commit (which currently doesn't compile).

Comment (FlatOutPS2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: I move it back(this appearing in the wrong commit was due to git messing with the PR). Seems like the commit itself was fine, as that was the only change I made and it just finished compiling succesfully.

File: pcsx2/gui/Panels/MemoryCardListPanel.cpp [hunk 6]
```diff
@@ -1016,29 +1012,27 @@ void Panels::MemoryCardListPanel_Simple::OnOpenItemContextMenu(wxListEvent& evt)
...
+	if (idx != wxNOT_FOUND) {
 		const McdSlotItem& card( GetCardForViewIndex(idx) );
 
-		if (card.IsPresent){
-			junk->Append( McdMenuId_AssignUnassign,	card.Slot>=0?_("Eject card"):_("Insert card ...") );
-			junk->Append( McdMenuId_Duplicate,	_("Duplicate card ...") );
-			junk->Append( McdMenuId_Rename,		_("Rename card ...") );
-			junk->Append( McdMenuId_Create,		_("Delete card") );
+		if (card.IsPresent) {
+			junk->Append(McdMenuId_AssignUnassign,	card.Slot>=0?_("Eject card"):_("&Insert card ..."));
```
Comment (turtleli) [inline pcsx2/gui/Panels/MemoryCardListPanel.cpp]: Eject card doesn't have a shortcut?

Comment (FlatOutPS2) [reply pcsx2/gui/Panels/MemoryCardListPanel.cpp]: Fixed

**Pull request merged**